pkg/server/endpoints/node/handler.go

```diff
@@ -644,6 +674,13 @@ func (h *Handler) signCSRs(ctx context.Context,
 				return nil, err
 			}
 
+		} else if e := h.getDownstreamEntry(ctx, callerID); e != nil {
```

Comment:
There are two things that need to happen here: 1) know if the caller is authorized for downstream issuance, and 2) ensure that the CSR is actually intended to be a CSR for a CA certificate (vs a request for the downstream server's client SVID). The former should happen within the conditional here so we can reject the request if it's not authorized, and the latter would be the condition.

```go
else if spiffeID == h.c.TrustDomain.String() {
    entry, err := h.getDownstreamEntry(ctx, callerID)
    if err != nil {
        return nil, err
    }

    ...
```

Reply:
Fixed as per the suggestions.